### PR TITLE
Networking/chris/devel

### PR DESCRIPTION
**Pull request by Alramech**
ignore client/server/driver files

### CODE REVIEW

**Comment by yizow** [comment]
See comments on slack.

File: networking/config.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from enum import Enum
+
+class Commands(Enum):
+  ESTOP = "0"
+  START_GAME = "1"
+  TELEOP = "2"
+  END_GAME = "3"
```
Comment (yizow) [inline networking/config.py:7]: newline at end of file

File: networking/tentativename.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import lcm
```
Comment (yizow) [inline networking/tentativename.py:1]: Change the file name and choose a casing

File: networking/tentativename.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+def heartbeat_listen(lc, robot_number):
+  """
+  Subthread intended to listen for heartbeats
+  """
+  while True:
+    lc.handle()
+    
+t = threading.Thread(target = heartbeat_listen, args = (lc, 1))
+t.daemon=True
+t.start()
```
Comment (yizow) [inline networking/tentativename.py:45]: newline at end of file

File: networking/test.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import subprocess
+import time
+import filecmp
+import sys
+
+infilename = "test.py"
+
+outfile = open("testout.txt", "w")
```
Comment (yizow) [inline networking/test.py:8]: I recommend adding a `.gitignore` file in this directory to apply to various temp files youre going to be generating. such as lcm stuff.

File: networking/robot.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import lcm
+import threading
+import time
+from config import Commands as cmd
```
Comment (yizow) [inline networking/robot.py:4]: cmd is a python default module. 
I recommend not overwriting it.

If you maintain config to only be global constants and enum stuff, etc, I would recommend 
`from config import *`

File: networking/robot.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+class Robot():
+  """
+  Basic robot simulator. Features
+  game states as well as estop and basic i/o
+  """
+  def __init__(self, number):
+    self.number = number
+    self.teleop = False
```
Comment (yizow) [inline networking/robot.py:13]: `self.teleop` and `self.enabled` seem redundant.

File: networking/robot.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    self.motors = [0]*4
+    self.sensors = [True, 4]
+    self.enabled = False
+    self.estop = False
+        
+def heartbeat(lc, number):
+  """
+  Generates heartbeats for the robot
+  sends robot number as well as state flag
+  (0 = disabled, 1 = autonomous, 2 = teleop, 3 = estop)
```
Comment (yizow) [inline networking/robot.py:23]: i recommend having estop as 0, sinces it's more commonly used as an error state

File: networking/robot.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  """
+  i = 0
+  while True:
+    i += 1
+    state = robot.enabled + robot.teleop + robot.estop*2
```
Comment (yizow) [inline networking/robot.py:28]: what the heck is this supposed to mean?

File: networking/robot.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  if data == cmd.START_GAME:
+    robot.enabled = True
+
+  if data == cmd.TELEOP:
+    robot.teleop = True
+    
+  if data == cmd.END_GAME:
+    robot.enabled = False
+    
+lc = lcm.LCM()
```
Comment (yizow) [inline networking/robot.py:51]: probably want to choose a different name than `lc`

File: networking/robot.py [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+def cmd_handler(channel, data):
+  """
+  Handles recieved commands
+  """
+  data = cmd(data)
+  if data == cmd.ESTOP:
+    robot.enabled = False
+    robot.estop = True
+
+  if data == cmd.START_GAME:
```
Comment (yizow) [inline networking/robot.py:42]: Consider using `elif` for these just to catch bug cases where you alias enums together.

Comment (yizow) [reply networking/robot.py:42]: Another alternative is to use a dictionary mapping from enum -> function definitions.

File: networking/tentativename.py [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import lcm
+import threading
+import time
+from config import Commands as cmd
```
Comment (yizow) [inline networking/tentativename.py:4]: See above comments

File: networking/tentativename.py [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+def heartbeat_handler(channel, data):
+  """
+  Prints out heartbeat data
+  """
+  data = data.decode()
+  msg = data.split(' ')
+  print("Robot {}: state {}".format(msg[0], msg[1]))
+  
+lc = lcm.LCM()
```
Comment (yizow) [inline networking/tentativename.py:14]: Should probably move this to outside of function definitions

**Review by yizow — CHANGES_REQUESTED** [review]
PLS CHANGE FIXES PL0X

**Review by yizow — CHANGES_REQUESTED** [review]
Gitignore.

File: networking/robot.py [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+  Generates heartbeats for the robot
+  sends robot number as well as state flag
+  (0 = disabled, 1 = autonomous, 2 = teleop, 3 = estop)
+  """
+  i = 0
+  while True:
+    i += 1
+    state = robot.enabled + robot.teleop + robot.estop*2
+    status = "{} {}".format(robot.number, state)
+    lc.publish("ROBOT_HEARTBEAT", status)
```
Comment (nikitakit) [inline networking/robot.py:30]: I notice you're publishing strings on LCM.

I strongly recommend that you switch to publishing using LCM types instead (or another strict encoding system like protobuf) -- using types helps you keep your sanity when building a distributed system.

**Review by nikitakit — CHANGES_REQUESTED** [review]

File: networking/robot.py (hunk 9 above)
Comment (nikitakit) [reply networking/robot.py:51]: @yizow So `lcm` is already taken (module name), and you want something short...

**Pull request closed without merging**